### ISSUE TITLE
There is no foreign key inserted for one-to-one relationship

### ISSUE THREAD

**Issue by deepu105** [jhipster/generator-jhipster#1468]
When I generate a one-to-one between two entities, the create action doesnt have a dropdown to select the related entity, the foreign key column is generated but its not updated on create or update

For Example: I have a Bar entity which has a one-to-one with Foo, Bar is the owner and hence has foo_id as a column in Bar table, Now when i create a Bar Shouldn't I have an option to map a Foo to it?

Is this as per design or a bug?

Foo Entity
-------------
```

{
    "relationships": [
        {
            "relationshipId": 1,
            "relationshipName": "bar",
            "relationshipNameCapitalized": "Bar",
            "relationshipFieldName": "bar",
            "otherEntityName": "bar",
            "relationshipType": "one-to-one",
            "otherEntityNameCapitalized": "Bar",
            "ownerSide": false
        }
    ],
    "fields": [
        {
            "fieldId": 1,
            "fieldName": "name",
            "fieldType": "String",
            "fieldNameCapitalized": "Name",
            "fieldNameUnderscored": "name",
            "fieldInJavaBeanMethod": "Name",
            "fieldValidate": false
        }
    ],
    "fieldsContainOwnerManyToMany": false,
    "fieldsContainOneToMany": false,
    "fieldsContainLocalDate": false,
    "fieldsContainCustomTime": false,
    "fieldsContainBigDecimal": false,
    "fieldsContainDateTime": false,
    "fieldsContainDate": false,
    "changelogDate": "20150513072403",
    "pagination": "no",
    "validation": false
}
```

Bar Entity
-------------
```
{
    "relationships": [
        {
            "relationshipId": 1,
            "relationshipName": "foo",
            "relationshipNameCapitalized": "Foo",
            "relationshipFieldName": "foo",
            "otherEntityName": "foo",
            "relationshipType": "one-to-one",
            "otherEntityNameCapitalized": "Foo",
            "ownerSide": true
        }
    ],
    "fields": [
        {
            "fieldId": 1,
            "fieldName": "namee",
            "fieldType": "String",
            "fieldNameCapitalized": "Namee",
            "fieldNameUnderscored": "namee",
            "fieldInJavaBeanMethod": "Namee",
            "fieldValidate": false
        }
    ],
    "fieldsContainOwnerManyToMany": false,
    "fieldsContainOneToMany": false,
    "fieldsContainLocalDate": false,
    "fieldsContainCustomTime": false,
    "fieldsContainBigDecimal": false,
    "fieldsContainDateTime": false,
    "fieldsContainDate": false,
    "changelogDate": "20150513072509",
    "pagination": "no",
    "validation": false
}
```

My config is as below:
------------------------------

```
{
  "generator-jhipster": {
    "baseName": "test",
    "packageName": "com.foo.bar",
    "packageFolder": "com/foo/bar",
    "hibernateCache": "hazelcast",
    "clusteredHttpSession": "no",
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "h2Memory",
    "prodDatabaseType": "mysql",
    "useCompass": false,
    "buildTool": "gradle",
    "frontendBuilder": "grunt",
    "javaVersion": "8",
    "authenticationType": "session",
    "rememberMeKey": "---",
    "searchEngine": "no"
  }
}
```

**Comment by MathieuAA** [comment]
Hi,

First, I think you got the content of each file mixed up (shouldn't the first content be the one of Foo instead of Bar?).

Did you create Foo first? (if so, you should have been warned by JHipster).
You need to create Bar first, because Foo is dependent (Foo's the owner).

By the way, I think you may have incorrectly answered the questions.

"relationships": [
        {
            "relationshipId": 1,
            "relationshipName": "foo",
            "relationshipNameCapitalized": "Foo",
            "relationshipFieldName": "foo",
            "otherEntityName": "foo",
            "relationshipType": "one-to-one",
            "otherEntityNameCapitalized": "Foo",
            "ownerSide": true
        }
    ],

If this snippet I took from Foo (or was it Bar?) is correct, then you created a "reflexive" relationship for Foo (you should have answered Bar instead of Foo).

Try creating the Bar entity first, then Foo. Foo can remain the owner of the relationship.

**Comment by deepu105** [comment]
@MathieuAA : Yes thanks for pointing it out, I got the names mixed up, I have corrected the question. And Yes I did Foo first, but there was no warning coz I selected N for is this parent entity question. Anyway Let me try again as you said

**Comment by deepu105** [comment]
So I tried Again as you said, Hope its the correct way of doing it (Else the documentation really needs better explanation I guess)

But the outcome is the same, I still dont see the realtionship tagged while creating or dropdown, If you are able to generate a working sample can you post please. Not sure if Im reading the docs wrongly
```

yo jhipster:entity barAgain
The entity barAgain is being created.
Generating field #1
? Do you want to add a field to your entity? Yes
? What is the name of your field? name
? What is the type of your field? String
? Do you want to add validation rules to your field? No
=================BarAgain=================
name (String)
Generating field #2
? Do you want to add a field to your entity? No
=================BarAgain=================
name (String)
Generating relationships with other entities
? Do you want to add a relationship to another entity? Yes
? What is the name of the other entity? fooAgain
? What is the name of the relationship? fooAgain
? What is the type of the relationship? one-to-one
? Is this entity the owner of the relationship? No
===========BarAgain==============
name (String)
-------------------
fooAgain - fooAgain (one-to-one)
Generating relationships with other entities
? Do you want to add a relationship to another entity? No
===========BarAgain==============
name (String)
-------------------
fooAgain - fooAgain (one-to-one)
? Do you want pagination on your entity? No
Everything is configured, generating the entity...
   create .jhipster\BarAgain.json
   create src\main\java\com\foo\bar\domain\BarAgain.java
   create src\main\java\com\foo\bar\repository\BarAgainRepository.java
   create src\main\java\com\foo\bar\web\rest\BarAgainResource.java
   create src\main\resources\config\liquibase\changelog\20150513092223_added_entity_BarAgain.xml
   create src\main\webapp\scripts\app\entities\barAgain\barAgains.html
   create src\main\webapp\scripts\app\entities\barAgain\barAgain-detail.html
   create src\main\webapp\scripts\app\entities\barAgain\barAgain.js
   create src\main\webapp\scripts\app\entities\barAgain\barAgain.controller.js
   create src\main\webapp\scripts\app\entities\barAgain\barAgain-detail.controller.js
   create src\main\webapp\scripts\components\entities\barAgain\barAgain.service.js
   create src\test\java\com\foo\bar\web\rest\BarAgainResourceTest.java
   create src\test\gatling\simulations\BarAgainGatlingTest.scala
   create src\main\webapp\i18n\en\barAgain.json
   create src\main\webapp\i18n\fr\barAgain.json

yo jhipster:entity fooAgain
The entity fooAgain is being created.
Generating field #1
? Do you want to add a field to your entity? Yes
? What is the name of your field? name
? What is the type of your field? String
? Do you want to add validation rules to your field? No
=================FooAgain=================
name (String)
Generating field #2
? Do you want to add a field to your entity? No
=================FooAgain=================
name (String)
Generating relationships with other entities
? Do you want to add a relationship to another entity? Yes
? What is the name of the other entity? barAgain
? What is the name of the relationship? barAgain
? What is the type of the relationship? one-to-one
? Is this entity the owner of the relationship? Yes
===========FooAgain==============
name (String)
-------------------
barAgain - barAgain (one-to-one)
Generating relationships with other entities
? Do you want to add a relationship to another entity? No
===========FooAgain==============
name (String)
-------------------
barAgain - barAgain (one-to-one)
? Do you want pagination on your entity? No
Everything is configured, generating the entity...
   create .jhipster\FooAgain.json
   create src\main\java\com\foo\bar\domain\FooAgain.java
   create src\main\java\com\foo\bar\repository\FooAgainRepository.java
   create src\main\java\com\foo\bar\web\rest\FooAgainResource.java
   create src\main\resources\config\liquibase\changelog\20150513092320_added_entity_FooAgain.xml
   create src\main\webapp\scripts\app\entities\fooAgain\fooAgains.html
   create src\main\webapp\scripts\app\entities\fooAgain\fooAgain-detail.html
   create src\main\webapp\scripts\app\entities\fooAgain\fooAgain.js
   create src\main\webapp\scripts\app\entities\fooAgain\fooAgain.controller.js
   create src\main\webapp\scripts\app\entities\fooAgain\fooAgain-detail.controller.js
   create src\main\webapp\scripts\components\entities\fooAgain\fooAgain.service.js
   create src\test\java\com\foo\bar\web\rest\FooAgainResourceTest.java
   create src\test\gatling\simulations\FooAgainGatlingTest.scala
   create src\main\webapp\i18n\en\fooAgain.json
   create src\main\webapp\i18n\fr\fooAgain.json
```

barAgain
-----------
```
{
    "relationships": [
        {
            "relationshipId": 1,
            "relationshipName": "fooAgain",
            "relationshipNameCapitalized": "FooAgain",
            "relationshipFieldName": "fooAgain",
            "otherEntityName": "fooAgain",
            "relationshipType": "one-to-one",
            "otherEntityNameCapitalized": "FooAgain",
            "ownerSide": false
        }
    ],
    "fields": [
        {
            "fieldId": 1,
            "fieldName": "name",
            "fieldType": "String",
            "fieldNameCapitalized": "Name",
            "fieldNameUnderscored": "name",
            "fieldInJavaBeanMethod": "Name",
            "fieldValidate": false
        }
    ],
    "fieldsContainOwnerManyToMany": false,
    "fieldsContainOneToMany": false,
    "fieldsContainLocalDate": false,
    "fieldsContainCustomTime": false,
    "fieldsContainBigDecimal": false,
    "fieldsContainDateTime": false,
    "fieldsContainDate": false,
    "changelogDate": "20150513092223",
    "pagination": "no",
    "validation": false
}
```

fooAgain
--------------
```
{
    "relationships": [
        {
            "relationshipId": 1,
            "relationshipName": "barAgain",
            "relationshipNameCapitalized": "BarAgain",
            "relationshipFieldName": "barAgain",
            "otherEntityName": "barAgain",
            "relationshipType": "one-to-one",
            "otherEntityNameCapitalized": "BarAgain",
            "ownerSide": true
        }
    ],
    "fields": [
        {
            "fieldId": 1,
            "fieldName": "name",
            "fieldType": "String",
            "fieldNameCapitalized": "Name",
            "fieldNameUnderscored": "name",
            "fieldInJavaBeanMethod": "Name",
            "fieldValidate": false
        }
    ],
    "fieldsContainOwnerManyToMany": false,
    "fieldsContainOneToMany": false,
    "fieldsContainLocalDate": false,
    "fieldsContainCustomTime": false,
    "fieldsContainBigDecimal": false,
    "fieldsContainDateTime": false,
    "fieldsContainDate": false,
    "changelogDate": "20150513092320",
    "pagination": "no",
    "validation": false
}
```

**Comment by MathieuAA** [comment]
Yes, you did it right!

Now, it is normal you don't see any drop-down (don't know if it's mentioned somewhere, gotta check).
You may have to do it yourself (so that you get what you want), and it's no big deal anyway because it's quite an easy thing to do.

Anyway, I'm gonna check the docs.

**Comment by MathieuAA** [comment]
I managed to get the drop-down (I didn't modify anything).
I created another entity 'Thing', which has a one-to-many relationship with another entity I created just after : 'Foo'. When I tried to create a new Foo, the drop-down was available for me to choose which Thing I wanted for my new Foo.

![drop-down example](https://cloud.githubusercontent.com/assets/2753381/7608223/def9b7e6-f967-11e4-93ee-4d8fa110a248.png)

**Comment by deepu105** [comment]
@MathieuAA : Yes I know it works for one-to-many/many-to-one combination. and yes it is fairly easy to modify and achieve in one-to-one. My intention of opening this was to check if this is intended behaviour. coz if JH generated the dropdown for one-to-many and many-to-one wouldnt it be ideal and complete if it does the same for one-to-one also, so that the generated entity is usable right out of the box?

**Comment by MathieuAA** [comment]
You can propose it as a PR if you have the time to do it.

**Comment by deepu105** [comment]
One more problem is When you update the one-to-one column in the Table like below, there comes the dreaded recursion error. Not sure if im doing it wrong

![image](https://cloud.githubusercontent.com/assets/1107223/7609976/5c8541b0-f9ab-11e4-8cb4-a4cb1bec56dc.png)

![image](https://cloud.githubusercontent.com/assets/1107223/7609985/79649380-f9ab-11e4-83e0-59697a341f08.png)


response: (Have clipped of repeating entries inbetween)
```
[{"id":1,"name":"test","barAgain":{"id":1,"name":"test","fooAgain":{"id":1,"name":"test","barAgain":{"id":1,"name":"test","fooAgain":{"id":1,"name":"test","barAgain":{"id":1,"name":"test","fooAgain":{"id":1,"name":"test","barAgain":{"id":1,"name":"test","fooAgain":{"id":1,"name":"test","barAgain":{"id":1,"name":"test","fooAgain":{"id":1,"name":"test","barAgain":{"id":1,"name":"test","fooAgain":{"id":1,"name":"test","barAgain":{"id":1,"name":"test","fooAgain":{"id":1,"name":"test","barAgain":
.
.
.
{"id":1,"name":"test","fooAgain":{"id":1,"name":"test","barAgain":{"id":1,"name":"test","fooAgain":{"id":{"timestamp":1431518449614,"status":200,"error":"OK","exception":"org.springframework.http.converter.HttpMessageNotWritableException","message":"Could not write content: Infinite recursion (StackOverflowError) (through reference chain: com.foo.bar.domain.BarAgain[\"fooAgain\"]->com.foo.bar.domain.FooAgain[\"barAgain\"]->com.foo.bar.domain.BarAgain[\"fooAgain\"]->com.foo.bar.domain.FooAgain[\"barAgain\"]->com.foo.bar.domain.BarAgain[\"fooAgain\"]-
....

>com.foo.bar.domain.FooAgain[\"barAgain\"]); nested exception is com.fasterxml.jackson.databind.JsonMappingException: Infinite recursion (StackOverflowError) (through reference chain: com.foo.bar.domain.BarAgain[\"fooAgain\"]->com.foo.bar.domain.FooAgain[\"barAgain\"]->com.foo.bar.domain.BarAgain[\"fooAgain\"]-
```

**Comment by deepu105** [comment]
@MathieuAA : I would love to do that, but still havent managed to find time to look at the source code and find my way around it. Will try to do though.

**Comment by jdubois** [comment]
I will have a look at this later this week, but just a quick note: one-to-one relationship are very rarely used (I never use them myself, I consider that if 2 entities have a one-to-one relationship, then there should only be one entity), so it's not as well-tested as the other kind of relationships.

**Comment by deepu105** [comment]
@jdubois You are correct, even I wouldnt prefer one-to-one in most cases. But when there is a Entity which needs to be reused in more than one place it would be one-to-one right, and in my case the data model is pretty much industry standard, say Flight > Aircraft etc, so Flight and Aircraft are different entities but will have a one2one relationship

**Comment by gzsombor** [comment]
I guess, one aircraft will have more than one flight. Or I dont want to be on that aircraft :)
But otherwise, sending circular graphs over json never work, so either you put some "@JsonIgnore" in one of the fields, or you start using DTOs and have a clear mapping strategy, how to solve it.

**Comment by deepu105** [comment]
@gzsombor : ha ha ideally yes, but in some use cases no(its complicated to explain so nevermind).

Hmm yes we are trying out some jackson annotations.

**Comment by jdubois** [comment]
After closer look: yes, it is "normal", in the sense that this hasn't been implemented. So you don't have drop-downs because I didn't code it, basically :-)
It shouldn't be hard to do, but I don't know if that's useful for a one-to-one relationship -> I guess that in the end you want everything in the same screen, not two separate screens

**Comment by deepu105** [comment]
yup guess you are right

**Comment by jdubois** [comment]
Yes you are correct, the JsonIgnore annotation is a real issue, I'll close this ticket with the correction

**Comment by andidev** [comment]
I came across the same issue. I'll try to implement a dropdown if it's not too much work.

**Comment by deepu105** [comment]
@andidev  I guess the drop down will be similar to the one to many dropdown but as @jdubois said it doesnt make sense to have it seperate and add as dropdown. It would be great if we can merge the one-to-one entity screens as a single form but that would be lot of work

**Comment by gmarziou** [comment]
one-to-one entity screens as a single form works most of the time and in particular I think it would address the concern that a lot of people have which is to extend User.

But how would it work if the relation is optional like for instance between a user and his passport.

**Comment by gmarziou** [comment]
in Rails, they have the notion of nested forms and nested attributes, it means being able to update several models with one form and request and yes it can be complex.

**Comment by andidev** [comment]
The single form solution seems the best way to do it but I don't have any time for that right now.

I'm thinking about simply adding a simple dropdown as it works for the many-to-one relationship. Isn't that at least better than the current solution where you are not able to set the relationship at all?

**Comment by andidev** [comment]
Could somebody review my PR #1757
It was a bit more tricky to make it work than I thought.
